ray-operator/api/raycluster/v1alpha1/raycluster_types.go

```diff
@@ -62,6 +65,18 @@ type ScaleStrategy struct {
 	WorkersToDelete []string `json:"workersToDelete,omitempty"`
 }
 
+// JobSpec is the spec for the job which will be submitted when raycluster is ready
+type JobSpec struct {
+	// Dict to setup execution environment, it is variable, the map value is expected a serialized value.
+	RuntimeEnv map[string]string `json:"runtimeEnv,omitempty"`
+	// Command to start execution, ex: "python script.py"
+	Entrypoint string `json:"entrypoint,omitempty"`
+	// Metadata to pass in to configure job behavior or use as tags
+	Metadata map[string]string `json:"metadata,omitempty"`
```

Comment:
https://github.com/ray-project/ray/blob/3cb85859cd4eb23a820871c3b00ba8215d6d59a1/dashboard/modules/job/sdk.py#L296-L301

above three fields should cover all options supported by job client. for the job_id, i feel currently upstream api doesn't give a well-defined name, we can use operator to generate one and explicitly pass into the client.